go/ct/st/transient_test.go

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package st
+
+import (
+	"reflect"
+	"testing"
+
+	. "github.com/Fantom-foundation/Tosca/go/ct/common"
+)
+
+func getTransientChanges() map[string]func(*Transient) {
+	return map[string]func(*Transient){
+		"set": func(t *Transient) {
+			t.SetStorage(NewU256(1), NewU256(17))
+		},
+		"remove-current": func(t *Transient) {
+			t.DeleteStorage(NewU256(42))
+		},
+	}
+}
+func TestTransient_Clone(t *testing.T) {
+	tests := getTransientChanges()
+
+	t1 := &Transient{}
+	t1.SetStorage(NewU256(42), NewU256(1))
+
+	for name, change := range tests {
+		t.Run(name, func(t *testing.T) {
+			t2 := t1.Clone()
+			if !reflect.DeepEqual(t1, t2) {
+				t.Errorf("Clone is broken, want %v, got %v", t1, t2)
+			}
+			change(t2)
+			if reflect.DeepEqual(t1, t2) {
```

Comment:
nit: if you want to go one extra step more, check can be be parametrized and then you could have all 3 tests in one, since the structures of the all 3 are the same, just the check changes.

Reply:
I think those functions are testing different things and it much easier to debug if they are separate.